sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs

```diff
@@ -57,11 +57,14 @@ public async Task InitVariables()
         }
 
         [Test]
-        [TestCase(AuthMethod.ConnectionString, "chat", TestName = "GettingTokenWithSingleScopeWithConnectionString")]
-        [TestCase(AuthMethod.KeyCredential, "chat", TestName = "GettingTokenWithSingleScopeWithKeyCredential")]
-        [TestCase(AuthMethod.TokenCredential, "chat", TestName = "GettingTokenWithSingleScopeWithTokenCredential")]
+        [TestCase(AuthMethod.ConnectionString, "chat", TestName = "GettingTokenWithSingleChatScopeWithConnectionString")]
+        [TestCase(AuthMethod.ConnectionString, "voip", TestName = "GettingTokenWithSingleVoIPScopeWithConnectionString")]
         [TestCase(AuthMethod.ConnectionString, "chat", "voip", TestName = "GettingTokenWithMultipleScopesWithConnectionString")]
+        [TestCase(AuthMethod.KeyCredential, "chat", TestName = "GettingTokenWithSingleChatScopeWithKeyCredential")]
+        [TestCase(AuthMethod.KeyCredential, "voip", TestName = "GettingTokenWithSingleVoIPScopeWithKeyCredential")]
         [TestCase(AuthMethod.KeyCredential, "chat", "voip", TestName = "GettingTokenWithMultipleScopesWithKeyCredential")]
+        [TestCase(AuthMethod.TokenCredential, "chat", TestName = "GettingTokenWithSingleChatScopeWithTokenCredential")]
+        [TestCase(AuthMethod.TokenCredential, "voip", TestName = "GettingTokenWithSingleVoIPScopeWithTokenCredential")]
```

Comment:
I wonder what's the purpose of the `AuthMethod`. Aren't all the tests run against all auth methods automatically in the pipelines?

Reply:
@petrsvihlik I didn't know that tests in pipelines run using all auth methods automatically. Do you have any links to docs or code to confirm this?
I've added a couple of new test cases but different auth methods were already being tested.

Reply:
@AikoBB do you know anything about automatic usage of all auth methods in pipelines? Shall we go with all the test cases or do I need to remove some of them (which ones in this case)?

Reply:
@petrsvihlik we have different methods to create a client. However, from what I see, a one client is instantiated per each test case, and it's usually a client created with a connection string. Only one parametrized test used to be created with different Auth methods. My suggestion is that it was considered enough to test that clients created with different option, produce the same correct test result.
![image](https://user-images.githubusercontent.com/110589094/212026340-ff4dadd5-6193-443d-bd7c-d31f0fec050a.png)

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/communication/Azure.Communication.Identity/tests/Infrastructure/CommunicationIdentityClientLiveTestBase.cs#L35-L54

As for the purpose of `AuthMethod`, it's not possible to call a method to create a client in test parameters because of CS0120 error  (An object reference is required for the nonstatic field, method, or property 'member'), so the indication to which credential to use when creating a client is passed.

Reply:
> . However, from what I see, a one client is instantiated per each test case, and it's usually a client created with a connection string. Only one parametrized test used to be created with different Auth methods.

@mjafferi-msft have you captured this in your research? afaik, in JS, we test all methods using all types of auth. perhaps it'd be good to do that for .net as well. it'd probably be good to solve it in a more generic way than adding an attribute with AuthMethod to every test method. Perhaps we could somehow extend the text matrix? Perhaps using the Azure.Core.TestFramework.TestEnvironment? I don't know...

> As for the purpose of AuthMethod, it's not possible to call a method to create a client in test parameters because of CS0120 error (An object reference is required for the nonstatic field, method, or property 'member'), so the indication to which credential to use when creating a client is passed.

sure, this is alright.


then I suggest the following:

can you please extract this

```
CommunicationIdentityClient client = authMethod switch
            {
                AuthMethod.ConnectionString => CreateClientWithConnectionString(),
                AuthMethod.KeyCredential => CreateClientWithAzureKeyCredential(),
                AuthMethod.TokenCredential => CreateClientWithTokenCredential(),
                _ => throw new ArgumentOutOfRangeException(nameof(authMethod)),
            };
```
as a separate method `protected CommunicationIdentityClient CreateClient(AuthMethod authMethod = AuthMethod.ConnectionString)` into the CommunicationIdentityClientLiveTestBase and make the other methods private? + replace the usages properly?